syntax = "proto3";

package obby.filewatcher;

option go_package = "github.com/obby/file-watcher/proto/generated";

service FileWatcher {
  // Start watching specified directories
  rpc StartWatching(WatchRequest) returns (WatchResponse);

  // Stop watching
  rpc StopWatching(StopRequest) returns (StopResponse);

  // Stream file events
  rpc StreamEvents(EventRequest) returns (stream FileEvent);

  // Update ignore/watch patterns
  rpc UpdatePatterns(PatternUpdate) returns (PatternResponse);
}

message WatchRequest {
  repeated string watch_paths = 1;
  repeated string ignore_patterns = 2;
  int32 debounce_ms = 3;
}

message WatchResponse {
  bool success = 1;
  string error = 2;
}

message StopRequest {
  // Empty for now, can add filters later
}

message StopResponse {
  bool success = 1;
  string error = 2;
}

message EventRequest {
  // Empty for now, can add filters later
}

message FileEvent {
  string path = 1;
  string event_type = 2; // created, modified, deleted, renamed
  int64 timestamp = 3;
  string old_path = 4; // for rename events
}

message PatternUpdate {
  repeated string watch_patterns = 1;
  repeated string ignore_patterns = 2;
}

message PatternResponse {
  bool success = 1;
  string error = 2;
}

